packages/elements/src/icon/index.ts

```diff
@@ -185,27 +154,35 @@ export class Icon extends BasicElement {
     this.iconReady = new Deferred<void>();
   }
 
-  /**
-   * Helper method, used to set the icon src.
-   * @returns {void}
-   */
-  private async setIconSrc (): Promise<void> {
-    // keep `src` in-sync with `icon` so that icon svg would be resolved after every `icon` update
-    this.src = this.icon ? await IconLoader.getSrc(this.icon) : this.icon;
+  private updateRenderer (value: string | null) {
+    if (value) {
+      if (this.iconMap && isBase64svg(this.iconMap)) {
+        void this.loadAndRenderIcon(this.iconMap);
+      }
+      else if (isUrl(value) || IconLoader.isPrefixSet) {
+        void this.loadAndRenderIcon(value);
+      }
+      else {
+        void this.loadAndRenderIcon(value, true);
+      }
+    }
+    else {
+      this.clearIcon();
+    }
   }
-
   /**
    * Tries to load an icon from the url provided
    * and the renders this into the icon template.
    * @param src Source location of the svg icon.
+   * @param isSprite is the sprite icon
    * @returns {void}
    */
-  private async loadAndRenderIcon (src: string): Promise<void> {
+  private async loadAndRenderIcon (src: string, isSprite = false): Promise<void> {
```

Comment:
I am not sure adding a flag is the best option, as it instantly makes the behaviour ambiguous